modules/platforms/dotnet/Apache.Ignite/IgniteServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,167 @@
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Apache.Ignite;
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+/// <summary>
+/// Extension method for setting up Apache Ignite services in an <see cref="Microsoft.Extensions.DependencyInjection.IServiceCollection" />.
+/// </summary>
+public static class IgniteServiceCollectionExtensions
+{
+    /// <summary>
+    ///  Registers an <see cref="IgniteClientGroup" /> and an <see cref="IIgnite" />.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+    /// <param name="configuration">
+    /// <see cref="IgniteClientGroupConfiguration" /> instance.
+    /// </param>
+    /// <param name="clientGroupLifetime">
+    /// The lifetime with which to register the <see cref="IgniteClientGroup" /> service in the container.
+    /// Defaults to <see cref="ServiceLifetime.Singleton" />.
+    /// </param>
+    /// <returns>Original service collection to chain multiple calls.</returns>
+    public static IServiceCollection AddIgniteClientGroup(
+        this IServiceCollection services,
+        IgniteClientGroupConfiguration configuration,
+        ServiceLifetime clientGroupLifetime = ServiceLifetime.Singleton)
+    {
+        AddIgniteClientGroupCore(services, (_, _) => configuration, clientGroupLifetime);
+
+        return services;
+    }
+
+    /// <summary>
+    ///  Registers an <see cref="IgniteClientGroup" /> and an <see cref="IIgnite" />.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+    /// <param name="configure">
+    /// Function to build the <see cref="IgniteClientGroupConfiguration" />.
+    /// </param>
+    /// <param name="clientGroupLifetime">
+    /// The lifetime with which to register the <see cref="IgniteClientGroup" /> service in the container.
+    /// Defaults to <see cref="ServiceLifetime.Singleton" />.
+    /// </param>
+    /// <returns>Original service collection to chain multiple calls.</returns>
+    public static IServiceCollection AddIgniteClientGroup(
+        this IServiceCollection services,
+        Func<IServiceProvider, IgniteClientGroupConfiguration> configure,
```

Comment:
This is usually an `Action` of some sort, examples:
* `AddRedis(ISignalRServerBuilder, Action<RedisOptions>)`
* `IServiceCollection AddNpgsqlSlimDataSource(IServiceCollection, string connectionString, Action<NpgsqlSlimDataSourceBuilder>)`